sdk/core/core/inc/az_http_transport.h

```diff
@@ -35,7 +35,7 @@ AZ_INLINE az_http_method az_http_method_patch() { return AZ_SPAN_FROM_STR("PATCH
  *     - `index` is out of range.
  */
 AZ_NODISCARD az_result
-az_http_request_get_header(_az_http_request* p_request, int32_t index, az_pair* out_result);
+az_http_request_get_header(_az_http_request const* p_request, int32_t index, az_pair* out_result);
```

Comment:
I think Jeff wanted to remove all `const` from parameters

Reply:
I did at one point but I'm OK if we have "const *" parameters

Reply:
Its odd to do it randomly.  If we want const parameters we should do that everywhere they are const.  Lets not do that at this time however

Reply:
I did this everywhere in IoT based on @JeffreyRichter 's recommendation :).

Reply:
That constness is different.
It makes little sense to write:
`f(int const n)`
`f(char* const n)`

But does make sense, and we should use as often as possible:
`f(char const* n)`

Reply:
We should not be removing const in the `foo const*` cases, absolutely not. I am not going to revert this change, and if we overdid this in the past due to misunderstanding - that was a mistake, and we should review our code and add back constness to the objects that pointers are pointing to. Maybe I can find some automatic way of doing that - when I would add constness everywhere and then start removing it until the code compiles.
It would be good to combine that with applying our new naming conventions - `ref_`s and `out_`s. (and no `p_`s).

I am basically consider this PR feedback as 'done', and not going to do more commits,and I'm going to resolve this conversation. Please revert it or ping me if you disagree and think that I should not check this in.